components/org.wso2.carbon.crypto.provider/src/main/java/org/wso2/carbon/crypto/provider/KeyStoreBasedInternalCryptoProvider.java

```diff
@@ -149,6 +164,32 @@ public KeyStoreBasedInternalCryptoProvider(KeyStore keyStore, String keyAlias, S
         }
     }
 
+    @Override
+    public byte[] encrypt(byte[] cleartext, String algorithm, String javaSecurityAPIProvider,
+                          boolean returnSelfContainedCipherText) throws CryptoException {
+
+        byte[] encryptedKey = encrypt(cleartext, algorithm, javaSecurityAPIProvider);
+        if (returnSelfContainedCipherText) {
+            Certificate certificate = null;
+            try {
+                certificate = getCertificateFromStore();
+                encryptedKey = createSelfContainedCiphertext(encryptedKey, algorithm, certificate);
+            } catch (KeyStoreException | CertificateEncodingException | NoSuchAlgorithmException e) {
+                String errorMessage = String.format("An error occurred while decrypting using the algorithm : '%s'"
+                        , algorithm);
```

Comment:
Shall we add provider info to this context

Reply:
Also the log is wrong right. It's an encryption

Reply:
Fixed with fde5a4d8773352de1654212bf563a4737bd658b2.
Yes, the log should be encryption.